### PR TITLE
New CRD based on recent observations

### PR DESCRIPTION
**Pull request by umangachapagain**
Issue: #34 
Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

### CODE REVIEW

File: deploy/crds/alerts_v1alpha1_storagealert_cr.yaml [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
+  namespace: storage-monitoring
+spec:
+  storage:
+  - provider: "nooba"
+    version:
+    - "v1.0"
+    namespace: "openshift-storage"
+  prometheus:
+    namespace: "openshift-monitoring"
+    label: "prometheus:k8s"
```
Comment (aruniiird) [inline deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:29]: label: "prometheus:k8s" // ? trying to understand the label here

File: deploy/crds/alerts_v1alpha1_storagealert_cr.yaml [hunk 2]
```diff
@@ -1,11 +1,48 @@
 apiVersion: alerts.monstorak.org/v1alpha1
 kind: StorageAlert
```
Comment (cloudbehl) [inline deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:2]: Why are we creating 3 cr ? for two monitoring projects?  why can't we have single(https://github.com/monstorak/monstorak/pull/32/files#diff-b81715432dc97f582ba99e9d4dadc3a7R31) cr which hold all the monitoring

File: deploy/crds/alerts_v1alpha1_storagealert_cr.yaml (hunk 1 above)
Comment (aruniiird) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:29]: Will there be only one prometheus instance for multiple storages?

File: deploy/crds/alerts_v1alpha1_storagealert_cr.yaml (hunk 2 above)
Comment (shtripat) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:2]: I agree with @cloudbehl here. We can have something like below

```
apiVersion: alerts.monstorak.org/v1alpha1
kind: StorageAlert
metadata:
  name: ceph-storagealert
  namespace: storage-monitoring
spec:
  storage:
  - provider: "ceph"
    version:
    - "v14.2.1-20190430"
    namespace: "openshift-storage"
  - provider: "noobaa"
    version:
    - "xxxxxx"
    namespace: "openshift-storage"
  prometheus:
    namespace: "openshift-monitoring"
    label: "prometheus:k8s"
```

Comment (shtripat) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:2]: That's what we meant initially while designing the CR.

File: deploy/crds/alerts_v1alpha1_storagealert_cr.yaml (hunk 1 above)
Comment (umangachapagain) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:29]: >label: "prometheus:k8s" // ? trying to understand the label here

That is the label of the prometheus instance to be selected.

> Will there be only one prometheus instance for multiple storages?

We will have one prometheus for multiple storages. But, you can create another object that makes use of another instance of prometheus (But, this is not suggested)

File: deploy/crds/alerts_v1alpha1_storagealert_cr.yaml (hunk 2 above)
Comment (umangachapagain) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:2]: > I agree with @cloudbehl here. We can have something like below
> 
> ```
> apiVersion: alerts.monstorak.org/v1alpha1
> kind: StorageAlert
> metadata:
>   name: ceph-storagealert
>   namespace: storage-monitoring
> spec:
>   storage:
>   - provider: "ceph"
>     version:
>     - "v14.2.1-20190430"
>     namespace: "openshift-storage"
>   - provider: "noobaa"
>     version:
>     - "xxxxxx"
>     namespace: "openshift-storage"
>   prometheus:
>     namespace: "openshift-monitoring"
>     label: "prometheus:k8s"
> ```

That is exactly what we have.

Comment (umangachapagain) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:2]: > Why are we creating 3 cr ? for two monitoring projects? why can't we have single(https://github.com/monstorak/monstorak/pull/32/files#diff-b81715432dc97f582ba99e9d4dadc3a7R31) cr which hold all the monitoring

If you look closely, these are different ways to create a CR.
You can create as many CRs as required. These are the ways you can do it.

Comment (shtripat) [reply deploy/crds/alerts_v1alpha1_storagealert_cr.yaml:2]: Ok.

**Pull request merged**